### ISSUE TITLE
Strict Argument Type

### ISSUE THREAD

**Issue by bhsmither** [cubecart/v6#2982]
PHP's `ctype_alnum()`, see:
```
http://docs.php.net/manual/en/function.ctype-alnum.php
```
there is a note concerning the integer value of the argument being treated in a certain way.

A lot of integers go through `Database->where()` that hit `ctype_alnum()`.

Suggest casting the argument to a string. (PHP 8.1 promises to treat any type as a string.)

**Comment by bhsmither** [comment]
Also `ctype_digit()` in `Database->pagination()`.

**Comment by bhsmither** [comment]
In `cubecart.class.php`, near line 854, there is:
```
httpredir('?'.http_build_query($query, null, '&'));
```
Parameter#2 requires a string. Suggest:
```
httpredir('?'.http_build_query($query, '', '&'));
```

**Comment by abrookbanks** [comment]
Fixed as part of #2708 already.

**Comment by abrookbanks** [comment]
Thanks.
